sky/adaptors/aws.py

```diff
@@ -2,13 +2,18 @@
 
 Thread safety notes:
 
-The results of session(), resource(), and client() are cached by each thread
-in a thread.local() storage. This means using their results is completely
-thread-safe.
+The results of session() is cached by each thread in a thread.local() storage.
+This means using their results is completely thread-safe.
 
 Calling them is thread-safe too, since they use a lock to protect
 each object's first creation.
 
+We do not cache the resource/client objects, because the credentials will be
+automatically rotated, but the cached resource/client object may not refresh the
+credential quick enough, which can cause unexpected NoCredentialsError. By
+creating the resource/client object every time, the credentials will be
+explicitly refreshed.
```

Comment:
```suggestion
We do not cache the resource/client objects, because some credentials may be
automatically rotated, but the cached resource/client object may not refresh the
credential quick enough, which can cause unexpected NoCredentialsError. By
creating the resource/client object from the thread-local session() object every time, the credentials will be
explicitly refreshed.
```

Reply:
Also, move this above para L8?  Then we can change L8 to say "Calling session(), resource(), and client() is thread safe, ..."